### PR TITLE
LabelGroup

### LINKED ISSUE #562: 🔮[New Component] LabelGroup
LabelGroup - Just like there is a ButtonGroup which sets the styles when Buttons should be rendered together, we should think about something similar for Labels.
In addition to the margin between the labels, this should also flex-wrap. Here is an example of the behavior today:
image

### PR DESCRIPTION
**Pull request by emplums**
Adds the LabelGroup component ✨ 

Closes #562 

#### Merge checklist
- [x] Tests written
- [x] Add or update TypeScript definitions (`index.d.ts`) if necessary
- [x] Docs written

### CODE REVIEW

File: docs/content/LabelGroup.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+The LabelGroup component is used to add commonly used margins and wrapping for groups of Labels.
+
+## Default example
+
+```jsx live
+<LabelGroup>
+  <Label m={1}>Default label</Label>
+  <Label m={1} scheme="gray-darker">Darker gray label</Label>
+  <Label m={1} outline>Default outline label</Label>
+</LabelGroup>
```
Comment (dmarcey) [inline docs/content/LabelGroup.md]: Can we get rid of the `m={1}` in the example?

Comment (emplums) [reply docs/content/LabelGroup.md]: Oh yeah 😂

File: docs/content/Label.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ The Label component is used to add contextual metadata to a design. Visually it
 
 ```jsx live
 <Label m={1}>Default label</Label>
-<Label m={1} scheme="gray-darker">Darker gray label</Label>
```
Comment (emplums) [inline docs/content/Label.md]: This `scheme` prop doesn't exist anymore 😬

File: src/__tests__/__snapshots__/LabelGroup.js.snap [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  color: #fff;
+  border-radius: 2px;
+  font-size: 12px;
+  padding-left: 3px;
+  padding-right: 3px;
+  padding-top: 4px;
+  padding-bottom: 4px;
+  margin: 4px;
+  background-color: #6a737d;
+  width: medium;
```
Comment (emplums) [inline src/__tests__/__snapshots__/LabelGroup.js.snap]: This `size: medium` style is due to a sort of bug @BinaryMuse and I discovered which she is currently working on a patch for. It's been happening for any component who uses a prop called `size` and doesn't have any side-effects so I'm fine merging this before Michelle's fix is out (though her fix will probably end up in the `release-14.4.0` branch as well

**Comment by emplums** [comment]
Looks like linting is failing due to `jsx-a11y` thinking these components are `<label>` elements 😬 https://github.com/evcohen/eslint-plugin-jsx-a11y/issues/638

**Comment by dmarcey** [comment]
> Looks like linting is failing due to `jsx-a11y` thinking these components are `<label>` elements 😬 [evcohen/eslint-plugin-jsx-a11y#638](https://github.com/evcohen/eslint-plugin-jsx-a11y/issues/638)

Looks like this actually comes from this line in `eslint-plugin-github`: `components: ['Label'],` (https://github.com/github/eslint-plugin-github/blob/40f8d9adab825057ae76e9f53f34ab2e29c0c3a9/lib/configs/react.js#L43).

If I get rid of `Label` in the components list, it works:

```
{
  "extends": ["plugin:github/es6", "plugin:github/react", "plugin:jsx-a11y/recommended"],
  "rules": {
    "import/no-namespace": 0,
    "no-unused-vars": [
      "error",
      {
        "ignoreRestSiblings": true
      }
    ],
    "jsx-a11y/label-has-for": [
      2,
      {
        "components": []
      }
    ]
  }
}```

**Pull request merged**